### PR TITLE
fix(desktop): surface legacy-dashboard banner when v2 is not active

### LINKED ISSUE #5498: [bug] Silent fallback to legacy v1 dashboard when localStorage is lost — presents as "all my worktrees disappeared"
## Summary

For accounts created before the v2-only cutoff, the v2 dashboard is gated on a **local-only** flag: the zustand store persisted at localStorage key `v2-local-override-v2` (`useIsV2CloudEnabled` → `optInV2 ?? (v2Only || dev)`). If localStorage is lost — quota-recovery wipe, profile corruption, moving machines — the app **silently reverts to the legacy v1 dashboard** with zero indication that anything changed.

## Why this is confusing

The v1 sidebar shows only legacy `local.db` projects with their "default" workspaces. Meanwhile the entire v2 world keeps working underneath (Electric sync runs, the CLI creates worktree workspaces, the sidebar reconciler happily writes placement rows) — but none of it is visible. To the user this presents as **"all my worktrees disappeared"**, which is hard to distinguish from data loss.

We hit this after recovering from the QuotaExceededError freeze (filed separately): recovery requires wiping the partition's Local Storage, which also nukes `v2-local-override-v2`. Diagnosing that the app was rendering v1 routes (`#/workspace/<local.db-id>`) instead of a v2 data bug took hours — nothing in the UI or logs says "you are on v1".

## Suggested fixes (any of these would do)

- Persist the v2 opt-in server-side (per user), not only in renderer localStorage.
- On fallback, show a banner: "You're viewing the legacy dashboard — switch back in Settings → Experimental" (the existing `V2AvailableBanner` is dismissible and easy to miss).
- Log the active dashboard version prominently at startup.

**Workaround for anyone hitting "my worktrees disappeared" after a localStorage wipe:**
`localStorage.setItem("v2-local-override-v2", JSON.stringify({state:{optInV2:true},version:0}))` + reload, or Settings → Experimental → re-enable v2.

**Environment:** desktop v1.13.1, macOS arm64, account predating V2_ONLY_USER_CUTOFF.

### PR DESCRIPTION
**Pull request by andyst-dev**
Fixes #5498

## Bug

For accounts created before `V2_ONLY_USER_CUTOFF`, the v2 dashboard is gated on a **renderer-local-only** flag: `optInV2`, persisted in zustand at localStorage key `v2-local-override-v2`. If localStorage is lost — quota-recovery wipe, profile corruption, moving machines — `optInV2` reverts to `null` and the app **silently drops the user back to the legacy v1 dashboard**. The v1 sidebar shows only legacy `local.db` projects, which presents as "all my worktrees disappeared" — indistinguishable from data loss. Nothing in the UI says "you are on v1".

## Fix

- **`deriveIsV2CloudEnabled.ts`** (new, pure): the `optInV2 ?? (isV2OnlyUser || isDev)` derivation extracted from `useIsV2CloudEnabled` so the fallback semantics are unit-testable, including the exact #5498 repro (opt-in → wipe → silent v1).
- **`V2AvailableBanner`**: rewritten with explicit copy — badge **"Legacy"**, title **"You're viewing the legacy dashboard"**, description pointing to **Settings → Experimental**, action **"Switch to v2"**. Previously the card said "Superset v2 is here / Try new version of Superset", which never told a user already on v2-recovering-from-a-wipe that they had fallen back.
- **`WorkspaceSidebar`** (v1): mounts `<V2AvailableBanner />` (only when not collapsed), so the legacy dashboard actually surfaces the banner — the component previously existed but was mounted nowhere.

The dismissible store is kept for the invitation case; the explicit legacy copy + Settings link addresses the diagnosis cost the issue calls out ("took hours — nothing in the UI or logs says 'you are on v1'").

## Validation

- `bun test` on `deriveIsV2CloudEnabled.test.ts`: **5/5 pass** (opt-in wins, opt-out wins, v2-only default, dev default, and the #5498 repro: opt-in → wipe → silent v1)
- `tsc --noEmit` on apps/desktop: **0 errors**
- Biome: clean
- Manual: on a pre-cutoff account with no v2 opt-in, the v1 sidebar now shows the Legacy card linking to Settings → Experimental.

---
## Summary by cubic
Surfaces a “Legacy dashboard” banner in the v1 sidebar whenever v2 isn’t active so localStorage-loss fallbacks are visible. Previously, losing `v2-local-override-v2` dropped pre‑cutoff users to v1 with no signal; now v1 labels “Legacy” and links to switch back. Explicit opt-outs show an informational card with no action. The banner auto‑re‑arms after v2 usage to catch later fallbacks.

- Mounts `V2AvailableBanner` in the v1 `WorkspaceSidebar` when the sidebar is not collapsed; hidden when v2 is active.
- Invitation state (no explicit opt-out): badge “Legacy”, title “You’re viewing the legacy dashboard”, action “Switch to v2” (Settings → Experimental); dismissible.
- Opt‑out state (`optInV2 === false`): non‑dismissible info card with no action.
- Resets the banner dismissal whenever v2 becomes active (`resetDismiss` in store).
- Extracts pure `deriveIsV2CloudEnabled` and updates `useIsV2CloudEnabled`; adds tests covering opt‑in/out, v2‑only/dev defaults, and the #5498 repro.
- Documents `v2-local-override-v2` and `v2-available-banner-v1` key lifecycles in `AGENTS.md`.

<sup>Written for commit 1e077b7d97c9c7af2db8b083df24c199b5e82d8b. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/superset-sh/superset/pull/6234?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Added a sidebar banner identifying the legacy dashboard and guiding eligible users to switch to v2 through Experimental settings.
  * Opted-out users now see a persistent informational card without an action.
  * Banner dismissal resets when v2 becomes active.

* **Bug Fixes**
  * Stabilized dashboard version selection for explicit preferences, v2-only users, and development environments.

* **Tests**
  * Added coverage for dashboard preferences, defaults, and regression scenarios.

### CODE REVIEW

**Comment by andyst-dev** [comment]
Thanks — both points are addressed in `8c322de`:

- **Stale dismissal hides fallback warning** (greptile P1): the banner now re-arms itself — whenever v2 is active and the dismissal is set, the dismissal is reset. So the sequence "dismiss → opt in to v2 → lose the independently-persisted v2 override" re-surfaces the legacy warning instead of hiding it behind the stale dismissal.
- **Documented v2-only opt-out state implemented** (coderabbit): `optInV2 === false` now renders a non-dismissible "opted out" card (no invitation action — the user chose legacy on purpose, including v2-only users). The dismissible invitation card (with the Settings → Experimental switch path) is kept only for the `optInV2 !== false` case (pre-cutoff or lost-override).

`tsc --noEmit`: 0 errors · `bun test`: 5/5 pass · Biome: clean.

**Comment by andyst-dev** [comment]
Thanks — addressed in `4fe5ff7`:

- **Persisted key lifecycle documentation** (coderabbit): added a "v2 dashboard keys" section to `apps/desktop/AGENTS.md` documenting bounds and deletion for both `v2-local-override-v2` (singleton opt-in/opt-out, dies with the v2/legacy split, loss = the #5498 silent-fallback case) and `v2-available-banner-v1` (singleton dismissal flag, auto-reset when v2 becomes active, `DEAD_KEYS` on banner removal).

The earlier comments (coderabbit 11:38 / cubic 11:45 about the opt-out branch) were on the previous head — the `optInV2 === false` non-dismissible branch was already implemented in `8c322de` (renders a card without `onAction`/`onDismiss` when the user explicitly opted out; the dismissible invitation card is kept only for `optInV2 !== false`).

`tsc --noEmit`: 0 errors · `bun test`: 5/5 pass · Biome: clean.

**Comment by andyst-dev** [comment]
Closing as superseded by an architecture refactor on main. The component this PR builds on (`V2AvailableBanner`) was removed in `one sidebar card at a time via SidebarCardSlot (#6733)`, and the team now ships headless `V1AutoMigration` (auto-flip to v2 on the v1 surface). Forcing my banner onto the new `SidebarCardSlot` system would fight that design — the legacy-v1 banner premise is being retired, so there is no clean target to rebase onto. No action needed.
